### PR TITLE
adding id to the state once uploading a new file

### LINKED ISSUE #1424: return Id on the upload file mutation
is required o overwrite supporting files

### PR DESCRIPTION
**Pull request by jyenes**
we have to add id in order to be able to replace the supporting file, once a new upload is coming with the same name 

closes #1424  
unblocks #1383

### CODE REVIEW

**Review by SemraSh — APPROVED** [review]
Looks good to me 👍

File: app/components/pages/SubmissionWizard/steps/Files/SupportingUpload.js [hunk 1]
```diff
@@ -185,11 +193,13 @@ class SupportingUpload extends React.Component {
...
   }
 
-  updateFileState = (fileId, newState) => {
+  updateFileState = (fileId, newState, id) => {
     const newFilesState = [...this.state.files]
     const fileIndex = newFilesState.findIndex(file => file.id === fileId)
     if (fileIndex > -1) {
-      newFilesState[fileIndex] = { ...newFilesState[fileIndex], ...newState }
+      const updatingState = { ...newFilesState[fileIndex] }
+      updatingState.file.id = id
```
Comment (diversemix) [inline app/components/pages/SubmissionWizard/steps/Files/SupportingUpload.js:201]: why do we pass in `id` ? ... How is it different from `fileId`

File: app/components/pages/SubmissionWizard/steps/Files/SupportingUpload.js [hunk 2]
```diff
@@ -168,10 +169,17 @@ class SupportingUpload extends React.Component {
       } else {
         result.value.upload
           .then(data => {
-            this.updateFileState(result.value.fileId, {
-              success: true,
-              loading: false,
-            })
+            const updatedFile = data.data.uploadSupportingFile.files.filter(
+              file => file.filename === result.value.filename,
```
Comment (diversemix) [inline app/components/pages/SubmissionWizard/steps/Files/SupportingUpload.js:173]: Isn't the `id` on `result` ?

File: app/components/pages/SubmissionWizard/steps/Files/SupportingUpload.js (hunk 1 above)
Comment (jyenes) [reply app/components/pages/SubmissionWizard/steps/Files/SupportingUpload.js:201]: id is the id from the dataBase, we need to pass it to store id within the state, fileId is the array id and it's being used to remove loading spinner, so basically to know wich row is being updated.

**Pull request merged**